packages/js/product-editor/src/blocks/product-fields/product-details-section-description/edit.tsx

```diff
@@ -0,0 +1,240 @@
+		( select ) => {
+			const { getBlockRootClientId } = select( 'core/block-editor' );
+			return getBlockRootClientId( clientId );
+		},
+		[ clientId ]
+	);
+
+	if ( ! rootClientId ) return;
+
+	function menuItemClickHandler(
+		productTemplate: ProductTemplate,
+		onClose: () => void
+	) {
+		return async function handleMenuItemClick() {
+			try {
+				await validate( productTemplate.productData );
+
+				await editEntityRecord(
+					'postType',
+					'product',
+					productId,
+					productTemplate.productData
+				);
+
+				await saveEditedEntityRecord< Product >(
+					'postType',
+					'product',
+					productId,
+					{
+						throwOnError: true,
+					}
+				);
+
+				createSuccessNotice(
+					__( 'Product type changed.', 'woocommerce' )
+				);
+			} catch ( error ) {
+				const message = getProductErrorMessage( error as WPError );
+				createErrorNotice( message );
+			}
+
+			onClose();
+		};
+	}
+
+	function resolveIcon( iconName?: string | null ) {
+		if ( ! iconName || ! ( iconName in icons ) ) return undefined;
+
+		const { Icon, [ iconName as never ]: icon } = icons;
+
+		return <Icon icon={ icon } size={ 24 } />;
+	}
+
+	function getMenuItem( onClose: () => void ) {
+		return function renderMenuItem( productTemplate: ProductTemplate ) {
+			const isSelected =
+				selectedProductTemplate?.id === productTemplate.id;
+			return (
+				<MenuItem
+					key={ productTemplate.id }
+					info={ productTemplate.description ?? undefined }
+					isSelected={ isSelected }
+					icon={
+						isSelected
+							? resolveIcon( 'check' )
+							: resolveIcon( productTemplate.icon )
+					}
+					iconPosition="left"
+					role="menuitemradio"
+					onClick={ menuItemClickHandler( productTemplate, onClose ) }
+					className={ classNames( {
+						'components-menu-item__button--selected': isSelected,
+					} ) }
+				>
+					{ productTemplate.title }
+				</MenuItem>
+			);
+		};
+	}
+
+	return (
+		<Fill name={ rootClientId }>
+			<div { ...blockProps }>
+				<p>
+					{ createInterpolateElement(
+						/* translators: <ProductTemplate />: the product template. */
+						__( 'This is a <ProductTemplate />.', 'woocommerce' ),
+						{
+							ProductTemplate: (
+								<span>
+									{ selectedProductTemplate?.title?.toLowerCase() }
+								</span>
+							),
+						}
+					) }
+				</p>
+
+				<Dropdown
+					focusOnMount={ false }
+					// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+					// @ts-ignore Property does exists
+					popoverProps={ {
+						placement: 'bottom-start',
+					} }
+					renderToggle={ ( { isOpen, onToggle } ) => (
+						<Button
+							aria-expanded={ isOpen }
+							variant="link"
+							onClick={ onToggle }
+						>
+							<span>
+								{ __( 'Change product type', 'woocommerce' ) }
+							</span>
+						</Button>
+					) }
+					renderContent={ ( { onClose } ) => (
+						<div className="wp-block-woocommerce-product-details-section-description__dropdown components-dropdown-menu__menu">
```

Comment:
Why are we setting the `components-dropdown-menu__menu` class name on this?

Reply:
This is a workaround to let the [Dropdown](https://wordpress.github.io/gutenberg/?path=/docs/components-dropdown--docs) menu work as a nested Dropdown. It does not support that currently. The design requirement for this is to use the [DropdownMenu v2](https://wordpress.github.io/gutenberg/?path=/docs/components-experimental-dropdownmenu-v2--docs) that supports such nesting thing but it is experimental and not included in the GB version we have installed so we can not use it for now until we upgrade to the new GB version that have that component.